### ISSUE TITLE
`PlatformMenuBar` application-level menu does not actually show the given `label` on macOS

### ISSUE THREAD

**Issue by GroovinChip** [flutter/flutter#104933]
When creating a `PlatformMenuBar` for a macOS application, the provided `label` property does not actually influence the text of the application-level<sup>1</sup> menu - it will always be the name the application was created with, unless it is manually changed in the macos runner.

<sup>1</sup> application-level menu here means the menu that represents the application itself, where the name of the application is used, the very first menu in the menu bar:
<img width="229" alt="Screen Shot 2022-05-29 at 1 04 44 PM" src="https://user-images.githubusercontent.com/4250470/170882568-96f41b7a-a596-4193-aceb-d22455f4e8dc.png">

## Steps to Reproduce

1. Execute `flutter run` on the code sample **_on macOS_** 
2. Observe that the name of the application shown in the menu bar is not as expected
3. Change the `label` property of the application to something completely different, and observe that it has no effect on the name shown in the menu

**Expected results:** 
I expect to see the text that I provide to `label` shown in the menu bar: "PlatformMenuBar Bug", or whatever other text I provide

**Actual results:** 
The text "platform_menu_bar_bug" remains unchanged no matter what text is provided to `label`

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return PlatformMenuBar(
      menus: [
        PlatformMenu(
          label: 'PlatformMenuBar Bug',
          menus: [
            PlatformMenuItemGroup(
              members: <MenuItem>[
                PlatformMenuItem(
                  label: 'About',
                  onSelected: () {},
                )
              ],
            ),
          ],
        ),
      ],
      body: MaterialApp(
        title: 'Flutter Demo',
        theme: ThemeData(
          primarySwatch: Colors.blue,
        ),
        home: const MyHomePage(title: 'Flutter Demo Home Page'),
      ),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

### `flutter run -d macos -v`
See [this pastebin](https://pastebin.com/CQRSb0aQ) for the log (GitHub wouldn't let me post with it, said the post was too long)

### `flutter analyze`
```
reubenturner@Reubens-MacBook-Pro-M1 platform_menu_bar_bug % flutter analyze
Analyzing platform_menu_bar_bug...                                      
No issues found! (ran in 1.2s)
```

### `flutter doctor -v`
```
reubenturner@Reubens-MacBook-Pro-M1 platform_menu_bar_bug % flutter doctor -v
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.4 21F79 darwin-arm, locale en-US)
    • Flutter version 3.0.1 at /Users/reubenturner/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (10 days ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/reubenturner/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 67.1.4
    • Dart plugin version 221.5591.58

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by danagbemava-nc** [comment]
Issue is reproducible on `stable` and `master` using the code sample provided above. 

<details>
<summary>video</summary>

https://user-images.githubusercontent.com/88313112/170963015-9f0761ad-f1fa-4839-b8d5-f3f9b1c2594b.mov
</details>


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.3.1 21E258 darwin-arm, locale en-GB)
    • Flutter version 3.0.1 at /Users/nexus/dev/sdks/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (10 days ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3.1 21E258 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```
[✓] Flutter (Channel master, 3.1.0-0.0.pre.1005, on macOS 12.3.1 21E258 darwin-arm, locale en-GB)
    • Flutter version 3.1.0-0.0.pre.1005 at /Users/nexus/dev/sdks/flutters
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c2e10541f3 (4 hours ago), 2022-05-30 00:18:09 -0400
    • Engine revision bb9926b0e6
    • Dart version 2.18.0 (build 2.18.0-151.0.dev)
    • DevTools version 2.13.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/nexus/Library/Android/sdk
    • Platform android-32, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3.1 21E258 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 102.0.5005.61

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by TahaTesser** [comment]
@GroovinChip 
This is expected, the first menus on macOS always have the app name. (Try any other macOS app). 

Adding another menu shows a custom label 

![Screenshot 2022-05-30 at 13 43 22](https://user-images.githubusercontent.com/48603081/170976089-84c569c3-7b50-412f-9e18-e3a35b8e0222.png)


```dart
    return PlatformMenuBar(
      menus: [
        PlatformMenu(
          label: 'PlatformMenuBar Bug',
          menus: [
            PlatformMenuItemGroup(
              members: <MenuItem>[
                PlatformMenuItem(
                  label: 'Test Menu',
                  onSelected: () {},
                )
              ],
            ),
          ],
        ),
        PlatformMenu(
          label: 'This works fine',
          menus: [
            PlatformMenuItemGroup(
              members: <MenuItem>[
                PlatformMenuItem(
                  label: 'Test Menu',
                  onSelected: () {},
                )
              ],
            ),
          ],
        ),
      ], 
```


We can document this :)

**Comment by GroovinChip** [comment]
@TahaTesser the [code sample in the API documentation](https://api.flutter.dev/flutter/widgets/PlatformMenuBar-class.html), as well as the ["What's new in Flutter 3.0"](https://medium.com/flutter/whats-new-in-flutter-3-8c74a5bc32d0) article, contradict your assertion - both of these indicate that the very first `PlatformMenuItem` will be the "application-level" menu (the very first entry in the menu bar) and will therefore update based on the label that is passed.

**Comment by TahaTesser** [comment]
@GroovinChip 
That's interestng, I've never seen the first menu label in any macOS app that isn't the app's name. 
Let me look if it is a regressuion.

**Comment by GroovinChip** [comment]
> @GroovinChip
> That's interesting, I've never seen the first menu label in any macOS app that isn't the app's name.
> Let me see if it is a regression.

Well that’s what I’m saying, really - we make our apps using dart-style package names, because otherwise our `pubspec.yaml` files will be invalid - and those package names are passed to the macos runners, which is what is used as the application-level menu in the menu bar, and I’m sure I’m not the only one who wants to be able to set the app name to be its properly formatted self in that place without having to edit the macos runner. Doing it via the `PlatformMenuBar` is much more convenient.

Basically, let’s say I create a Flutter app with package name `my_app`. I want the application-level menu bar item to say `My App` rather than `my_app`. See what I’m saying? The docs and the article indicate that I can do that with the `PlatformMenuBar`.

**Comment by TahaTesser** [comment]
> Basically, let’s say I create a Flutter app with package name `my_app`. I want the application-level menu bar item to say `My App` rather than `my_app`. See what I’m saying? The docs and the article indicate that I can do that with the `PlatformMenuBar`.

Good point!

**Comment by gjhoppy** [comment]
If anyone needs a workaround then open Runner,xcodeproj on Xcode and under Targets>Build Settings>Packaging you can set the Product Name and this will be used in the application menu. Setting different names for debug and release also works.

**Comment by driftwoodstudio** [comment]
Feb 2025: To be clear, this is still "broken" in the sense that the PlatformMenuBar class contains no documentation of the fact that the first item in PlatformMenuBar.menus[] list has it's label completely ignored. Either the native code being used, or MacOS itself, always uses the application name as the title of the first menu. Setting a custom label in Flutter for the first item does nothing, and the only way to change the title of the first item is to change the name of the application (as has been pointed out in earlier comments). 

In terms of "behave like MacOS" this behavior is the correct standard: Mac apps always use the app name as the title of the first menu. 

However, if this is strictly enforced by flutter and cannot be overridden, the PlatformMenuBar class documentation should be updated to reflect this fact.
